### DISCUSSION TITLE
[Laptop] How to get the secondary display to run on Nvidia, with the main display running on internal graphics

### DISCUSSION THREAD

**Discussion by clotodex** [qtile/qtile#4640] [Q&A]
My laptops Hdmi port is hardwired to a dedicated nvidia graphics card.
Now, mainly I want to run qtile on the iGpu (intel) due to power and driver reasons. However if I want to plug in an hdmi cable I currently need to restart qtile (and because of that all my programs) and relaunch it running on nvidia.

How would I go about setting this up on qtile? Is this possible? If not why and where would one need to implement?
Thank you :)

**Comment by jrwrigh** [comment]
Usually you don't need to restart applications to restart qtile. Are you having to shut down qtile, change a setting, then restart it?

**Reply by clotodex** [reply]
Yes, I need to change an environment variable for wlroots:
To start without nvidia (which i do normally): `__EGL_VENDOR_LIBRARY_FILENAMES=/usr/share/glvnd/egl_vendor.d/50_mesa.json dbus-run-session qtile start -b wayland`
To start with nvidia i just dont set the variable

**Reply by clotodex** [reply]
What env vars - or where do they need to be set - is qtile looking at on the reload?
